src/main/kotlin/frc/robot/robotstate/RobotTriggers.kt

```diff
@@ -54,11 +54,16 @@ fun bindRobotCommands() {
             .onTrue(driveToShootingPoint())
     }
     isIntaking.apply {
-        and(ballsEmpty).onTrue(startIntaking())
         and(hasFrontBall)
             .and(hasBackBall)
             .onTrue(roller.stop(), hopper.stop(), setShooting())
-        and(hasBackBall).and(hasFrontBall.negate()).onTrue(hopper.stop())
+        and(hasBackBall).and(hasFrontBall.negate()).apply {
+            onTrue(stopIntaking())
+            and(ballPoses::isNotEmpty).onTrue(roller.intake())
+        }
+        and(ballsEmpty)
+            .and(ballPoses::isNotEmpty)
+            .onTrue(roller.intake(), hopper.start())
```

Comment:
Perhaps it should be:

```suggestion
            and(ballPoses::isNotEmpty).whileTrue(roller.intake())
        }
        and(ballsEmpty)
            .and(ballPoses::isNotEmpty)
            .whileTrue(roller.intake(), hopper.start())
```
So the command will stop when the robot doesn't see balls?

Reply:
no its not like that the roller and the hopper will continue untill it stoped by the sensor